ops/testing.py

```diff
@@ -1890,6 +1890,42 @@ def run_action(self, action_name: str,
                 output=action_under_test.output)
         return action_under_test.output
 
+    def set_cloud_spec(self, content: Dict[str, str]):
+        """Set cloud specification (metadata) including credentials.
+
+        Call this method before trying to call `Harness.Model.get_cloud_spec`.
+
+        Example usage::
+
+            class TestCharm(unittest.TestCase):
+                def setUp(self):
+                    self.harness = ops.testing.Harness(MyVMCharm)
+                    self.addCleanup(self.harness.cleanup)
+
+                def test_start(self):
+                    self.harness.set_cloud_spec(
+                        {"name": "lxd", "type": "lxd", "endpoint": "https://127.0.0.1:8443"}
+                    )
+                    self.harness.begin_with_initial_hooks()
```

Comment:
The thing about this example is that it doesn't really do anything with the spec - I'm pretty sure if you took out the `begin_with_initial_hooks` it would still work, because no events are actually involved.

Maybe it's ok since this is just in the API docs and we don't want those to be huge, but could we have the charm with a trivial start handler that uses the spec? I don't think tests would ever want to get the spec, because the charm can't change it, so it'll only ever be what the test gave it.

Something like:

```python
class MyVMCharm(ops.CharmBase):
    ...
    def _on_start(self, event: ops.StartEvent):
        spec = self.model.get_cloud_spec()
        ....
```

And then remove the `expected=` and last two asserts from the test, having just a "..." or something instead (testing whatever the start would actually do with the spec).